server-channel.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+type ServerChannel interface {
```

Comment:
"ServerChannel" seems overly general, but it doesn't include the main RPC communications.  Maybe "ServerBackChannel" ? :P  Or "ServerSys"

How does this stuff mesh with the features TimC added for docker support?  (Maybe talk to him)